### PR TITLE
Bug 1863515 - Refactor navigation and add logs for address autofill UI tests

### PR DESCRIPTION
**Pull request by AndiAJ**
Summary:
- Refactored `fillAndSaveAddress` which contains the most used navigations and interactions from our address autofill UI tests and switched to using it where applicable.
- Added logs to every function in our `SettingsSubMenuAutofillRobot` as well as to other robots that interact with our address autofill tests

✅ All 10 affected address autofill UI tests successfully passed 75x on Firebase.

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1863515

### CODE REVIEW

**Review by jjSDET — APPROVED** [review]
LGTM

The only consideration I'd add is to move the data used in `fillAndSaveAddress` to a variable as your test data to re-use it more easily and set the stage for data-driven testing later.

**Review by sv-ohorvath — APPROVED** [review]
Nothing to add.

File: fenix/app/src/androidTest/java/org/mozilla/fenix/helpers/MatcherHelper.kt [hunk 1]
```diff
@@ -75,8 +80,10 @@ object MatcherHelper {
         for (appItem in appItems) {
             if (exists) {
                 assertTrue(appItem.waitForExists(waitingTime))
+                Log.i("MozTestLog", "assertItemContainingTextExists: Verified ${appItem.selector} exists")
             } else {
                 assertFalse(appItem.waitForExists(waitingTimeShort))
+                Log.i("MozTestLog", "assertItemContainingTextExists: Verified ${appItem.selector} does not exist")
```
Comment (AaronMT) [inline fenix/app/src/androidTest/java/org/mozilla/fenix/helpers/MatcherHelper.kt:86]: Can we declare the tag in the Class, as well as use `UI` in the tag?

e.g, private static final String TAG = "MozUITestLog";

**Review by AaronMT — CHANGES_REQUESTED** [review]

File: fenix/app/src/androidTest/java/org/mozilla/fenix/helpers/MatcherHelper.kt (hunk 1 above)
Comment (AndiAJ) [reply fenix/app/src/androidTest/java/org/mozilla/fenix/helpers/MatcherHelper.kt]: Thanks @AaronMT , ss discussed I've added it to [Constants](https://github.com/mozilla-mobile/firefox-android/pull/4390/files#diff-a219bf2265cd9fa8230ab386bf0e0a7a262fdcd3984dbccbf581d659e8f62efbR12)

**Review by AaronMT — APPROVED** [review]

**Pull request merged**